command/agent/node_pool_endpoint.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package agent
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+func (s *HTTPServer) NodePoolsRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	switch req.Method {
+	case "GET":
+		return s.nodePoolList(resp, req)
+	case "PUT", "POST":
+		return s.nodePoolUpsert(resp, req, "")
+	default:
+		return nil, CodedError(http.StatusMethodNotAllowed, ErrInvalidMethod)
+	}
+}
+
+func (s *HTTPServer) NodePoolSpecificRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	path := strings.TrimPrefix(req.URL.Path, "/v1/node/pool/")
+	switch {
+	default:
+		return s.nodePoolCRUD(resp, req, path)
+	}
+}
+
+func (s *HTTPServer) nodePoolCRUD(resp http.ResponseWriter, req *http.Request, poolName string) (interface{}, error) {
+	switch req.Method {
+	case "GET":
+		return s.nodePoolQuery(resp, req, poolName)
+	case "PUT", "POST":
+		return s.nodePoolUpsert(resp, req, poolName)
+	case "DELETE":
+		return s.nodePoolDelete(resp, req, poolName)
+	default:
+		return nil, CodedError(http.StatusMethodNotAllowed, ErrInvalidMethod)
+	}
+}
+
+func (s *HTTPServer) nodePoolList(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
```

Comment:
I think we can always use this?

```suggestion
func (s *HTTPServer) nodePoolList(resp http.ResponseWriter, req *http.Request) (any, error) {
```

Reply:
I keep forgetting about it 😬